lib/rules/no-loop-func.js

```diff
@@ -174,7 +174,7 @@ module.exports = {
          * - has a loop node in ancestors.
          * - has any references which refers to an unsafe variable.
          * @param {ASTNode} node The AST node to check.
-         * @returns {boolean} Whether or not the node is within a loop.
+         * @returns {void}
```

Comment:
This is incorrect. Should be boolean but we should update `return` on line 183 to `return false` to make it always return a boolean.

Reply:
Updating it on line 183 to return false still does not cause a boolean to be returned everywhere, as the function doesn't return elsewhere--i.e., it effectively returns `undefined` elsewhere too.

If this were a generic utility, it might make sense to return a boolean, but the `checkForLoops` function is currently only used as an ESLint rule callback and is not used for validation, so despite the name of the function, there is no benefit for it to return a boolean.

Reply:
Ah my bad, I misread the code, you're right.